agent/src/main/java/org/openjdk/jmc/agent/sap/boot/converters/SystemPropChangeLogger.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2025 SAP SE. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+package org.openjdk.jmc.agent.sap.boot.converters;
+
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.Properties;
+
+import org.openjdk.jmc.agent.sap.boot.util.ArgumentsHolder;
+import org.openjdk.jmc.agent.sap.boot.util.Arguments;
+import org.openjdk.jmc.agent.sap.boot.util.LoggingUtils;
+import org.openjdk.jmc.agent.sap.boot.util.OutputCommand;
+
+public class SystemPropChangeLogger {
+	private static final Properties systemProps = AccessController.doPrivileged(new PrivilegedAction<Properties>() {
```

Comment:
Could probably also be simplified to a one liner with a lambda, but up to you!

Reply:
Yes, with a proper type cast.